sys/vfs/vfs.c

```diff
@@ -954,14 +977,20 @@ bool vfs_iterate_mount_dirs(vfs_DIR *dir)
      * we'd need to let go of it now to actually open the directory. This
      * temporary count ensures that the file system will stick around for the
      * directory open step that follows immediately */
-    atomic_fetch_add(&next->open_files, 1);
+    uint16_t before = atomic_fetch_add_u16(&next->open_files, 1);
+    /* We cannot use assume() here, an overflow could occur in absence of
+     * any bugs and should also be checked for in production code. We use
+     * expect() here, which was actually written for unit tests but works
+     * here as well */
```

Comment:
```suggestion
    uint16_t before = atomic_fetch_add_u16(&next->open_files, 1);
    /* We cannot use assume() here, an overflow could occur when the only
     * bug present is a memory leak, and should also be checked for in
     * production code. We use expect() here, which was actually written for
     * unit tests but works here as well */
```

Changes after the first lines are just impact of (guessed) reflowing.

Yes, technically there can be an application where open files exceed 2^16 and still not leak memory, but the file handles alone would be 256KiB at the very least, so it's safe to assume that there is a leak.

Reply:
Fair enough. I'd still be confused reading this as to how this can happen without a bug (such as a memory leak), but no biggie.